### PR TITLE
Fixes #347 - Group hierarchy changes

### LINKED ISSUE #347: Apply changes related to the complex group hierarchy
### Tasks
- Manage > Users
  - Refresh or replace the screencast near the top with the new dialog layout (permissions + group permissions fields)
  - Reference Guide: User Details
    - Refresh the screenshot of the edit user dialog
    - Split **Permissions** into **Roles** and **Group permissions** and rename them accordingly
    - Explain that the **Group permissions** field visibility depends on the checkbox state of the _Agent_ role
- Manage > Groups > Group Settings
  - Add **Parent group** field and slot it between **Name** and **Assignment timeout**
    - Explain the logic behind the group tree: parent-child inheritance and how the current group will be hidden from this selection. Also touch the topic that children of the current group will not be shown in this tree too.
  - Refresh sample configuration screenshot near the end
- Manage > Groups > Group Access Levels
  - Refresh the screenshot near the top, but add similar groups as now to the list. Make sure the new add row is visible, in an empty state.
  - Refresh or replace the screencast with the Edit User dialog.
  - Refresh or replace the screenshot and screencast with the Edit Role dialog. Drop the ~~`user's`~~ moniker, if anything it should be `user`.
  - Remind the user that they are supposed to first choose which group they would like to configure and then to explicitly click on the green **+ Add** button so it's added to the list before saving.
  - Warn the user that the group permissions table will be shown only if `ticket.agent` permission is selected.
  - Refresh screenshots in the Example section, they should show the **Remove** button in its own column, but it's fine not to render the add new group row
- Manage > Roles > Agent Permissions
  - Refresh screenshot under `ticket.agent` permission
  - Explain that activating/deactivating this permission will show/hide the **Group permissions** field below the **Permissions** field
- Channels > Web
  - Refresh screenshot near the top to show the new tree select field for **Group selection for Ticket creation**
  - Update default value under **Group selection for Ticket creation** to `(empty selection)`
- Channels > Email > Signatures
  - Refresh screenshot of the edit group dialog so it shows the new **Parent group** field

### PR DESCRIPTION
**Pull request by ralf401**

### CODE REVIEW

**Comment by ralf401** [comment]
I changed the wording from "group access" to "group permission(s)" in manage > groups > group permissions (formerly Group Access Levels). It was changed in the UI as well.

File: manage/groups/settings.rst [hunk 1]
```diff
@@ -13,6 +13,18 @@ Name
...
+   You can define *another* group as a parent group here. This is useful if you
+   want to create group hierarchies. In such a hierarchy (child/parent
+   relation), the tree of the parent group is inherited to the child group. That
+   means if you move a group which itself has child groups, the child groups
+   are "moved" accordingly. However, there is no inheritance of membership or
+   permissions. That means you can treat each group as an individual group -
+   even if they have a child/parent relation.
+
+   Note: Zammad doesn't show you any child group of the currently selected
+   group. This would lead to a circular reference!
```
Comment (dvuckovic) [inline manage/groups/settings.rst:26]: Perhaps we can switch this into a `hint` or `info` block.

File: manage/users/index.rst [hunk 2]
```diff
@@ -161,29 +161,30 @@ The ones that aren’t are described below.
...
-      or can’t be assigned a ticket,
-      group access levels are likely to blame.
+Group Permissions
+   The :doc:`/manage/groups/access-levels` define which tickets
+   an agent can work with.
+   If an agent is not receiving notifications for incoming tickets
+   or can't be assigned to a ticket, you should have a look on the group
+   permissions.
+   Please note that the visibility of the group permission table depends on the
+   role selection - it has to be "Agent" that the table shows up.
```
Comment (dvuckovic) [inline manage/users/index.rst:176]: Not entirely true: it has to be a role that itself has a `ticket.agent` permission assigned.

File: manage/users/index.rst [hunk 3]
```diff
@@ -161,29 +161,30 @@ The ones that aren’t are described below.
...
 
    .. figure:: /images/manage/users/user-details-permissions.png
       :alt: Permissions in the edit user dialog
       :align: center
       :width: 70%
 
-      **Top:** A user’s roles decide what kinds of actions they can perform
-      and which :doc:`groups </manage/groups/index>` they belong to.
-      **Bottom:** Group assignments can alternately be set on a per-user basis.
+   **Top:** A user's roles decide what kinds of actions they can perform
```
Comment (dvuckovic) [inline manage/users/index.rst:183]: `A user's roles what kinds of actions` => `User's roles decide what kind of actions`

**Review by dvuckovic — APPROVED** [review]
Great job! 👏

**Comment by ralf401** [comment]
> Great job! 👏

Thank you!! :-D

**Pull request merged**